cluster-api-gcp/deploy/deploy_helper.go

```diff
@@ -293,3 +293,18 @@ func (d *deployer) waitForApiserver(master string, timeout time.Duration) error
 	}
 	return err
 }
+
+// expects new config as string and an existing file to merge new config into.
+func mergeConfigIfExists(newConfig, existingFilePath string) (*clientcmdapi.Config, error) {
+	// store newconfig in a temp file, existing api expects file paths.
+	tempFile, _ := ioutil.TempFile("", "")
+	defer os.Remove(tempFile.Name())
+	if _, err := tempFile.WriteString(newConfig); err != nil {
+		return nil, err
+	}
+	// new Config takes precedence
+	loadingRules := clientcmd.ClientConfigLoadingRules{
```

Comment:
It seems this doesn't update the current config (d.configPath) but rather store in a separate file and load it as well.

Will this work for further access to the cluster after bootstrapping? It seems that if we'll use d.configPath for kubectl and further access to the cluster, we should consider actually merging into the d.configPath configuration file.

Reply:
Config gets written to the d.configPath. Here it takes the new config which is for the cluster just created, writes it to a temp file. existingFilePath is config already present on the host. Load() call merges both the config together into a single config object which this function returns. 

The caller of this function then writes this returned config to d.configPath file.
I hope this answers your question.

Reply:
It does, thank you! I had missed that the caller writes it to the config.